### PR TITLE
Preserve next line with trailing comment

### LINKED ISSUE #526: Lost blank line because of comment
Input:

```js
function a() {
  const a = 5; // ms

  return a;
}
```

Output:

```js
function a() {
  const a = 5; // ms
  return a;
}
```

Expected:

```js
function a() {
  const a = 5; // ms

  return a;
}
```

(Remove the comment and the empty line will be preserved.)

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22function%20a()%20%7B%5Cn%20%20const%20a%20%3D%205%3B%20%2F%2F%20ms%5Cn%5Cn%20%20return%20a%3B%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by vjeux**
It turns that our `hasNextLine` logic needs to be tuned to skip all the trailing comments. The code is not pretty but it does the job. It looks like it fixes a bunch of things in the test cases :)

I made sure that nested inline comments are NOT valid JavaScript

```js
/* /* a */ */
Uncaught SyntaxError: Unexpected token *
```

so it is okay to do a dumb search for */ when you are in a comment

Fixes #526

### CODE REVIEW

File: tests/preserve_line/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const a = 5; /* comment */ /* comment */ // comment
+  return a;
+}
+
+function a() {
+  const a = 5; /* comment */ /* comment */ // comment
+
+  return a;
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (vjeux) [inline tests/preserve_line/__snapshots__/jsfmt.spec.js.snap]: This is the test I added

**Comment by jlongster** [comment]
Wow, that's an interesting edge case, I never thought of that! The implementation looks great to me, thanks!
